app/src/main/java/org/matrix/matrixandroidsdk/activity/HomeActivity.java

```diff
@@ -257,7 +259,13 @@ public boolean onChildClick(ExpandableListView parent, View v,
                     roomId = mAdapter.getPublicRoomAt(childPosition).roomId;
                 }
 
-                CommonActivityUtils.goToRoomPage(roomId, HomeActivity.this);
+                if (Intent.ACTION_SEND.equals(getIntent().getAction()) && getIntent().getType() != null) {
```

Comment:
Having thought about it a bit more, making another ``Activity`` may end up being a lot of work for this patch to do. I'd be happy with some visual indication that I was sharing something, and that I need to select a room to complete the action. Please also remember to kill the ``Intent`` (e.g. using ``activity.setIntent(null)``) after you have processed it, to avoid replaying it under certain ``Activity`` lifecycle changes.